src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -675,7 +686,12 @@ def __init__(
 @rule
 async def setup_pex_process(request: PexProcess, pex_environment: SandboxPexEnvironment) -> Process:
     pex = request.pex
-    argv = pex_environment.create_argv(pex.name, *request.argv, python=pex.python)
+    pex_bin = (
+        os.path.relpath(pex.name, request.working_directory)
+        if request.working_directory
+        else pex.name
+    )
+    argv = pex_environment.create_argv(pex_bin, *request.argv, python=pex.python)
```

Comment:
And ditto. If create_argv took in a working_directory it could adjust the argv0 appropriately and then all process setup adjusting is centralized.